packages/at_onboarding_cli/lib/src/onboard/at_onboarding_service_impl.dart

```diff
@@ -339,41 +329,48 @@ class AtOnboardingServiceImpl implements AtOnboardingService {
   ///If not, wait until secondary is created
   Future<void> _waitUntilSecondaryCreated() async {
     final maxRetries = 50;
-    int _retryCount = 0;
-    SecondaryAddress? _secondaryAddress;
+    int retryCount = 1;
+    SecondaryAddress? secondaryAddress;
     SecureSocket? secureSocket;
 
-    while (_retryCount < maxRetries && _secondaryAddress == null) {
-      logger.finer('retrying find secondary.......$_retryCount/$maxRetries');
+    while (retryCount <= maxRetries && secondaryAddress == null) {
+      await Future.delayed(Duration(seconds: 3));
+      logger.finer('retrying find secondary.......$retryCount/$maxRetries');
       try {
-        _secondaryAddress =
-        await _atLookup?.secondaryAddressFinder.findSecondary(_atSign);
+        secondaryAddress =
+            await _atLookup?.secondaryAddressFinder.findSecondary(_atSign);
       } on Exception catch (e) {
         logger.finer(e);
       }
-      _retryCount++;
+      retryCount++;
+    }
+
+    if (secondaryAddress == null){
+      logger.severe('Could not find secondary address for $_atSign');
```

Comment:
you can mention could not find secondary address after max retries

Reply:
addressed review comments as part of 3ecbf3191d2a90e5b21cdf9aa8b6da7421e78437